src/components/Trade/TradeConfirm/TradeConfirm.tsx

```diff
@@ -99,10 +102,24 @@ export const TradeConfirm = ({ history }: RouterProps) => {
     [bip44Params?.accountNumber, buyAssetAccountIds],
   )
 
-  const parsedBuyTxId = useMemo(
-    () => serializeTxIndex(buyAssetAccountId, buyTxid, trade?.receiveAddress ?? ''),
-    [buyAssetAccountId, trade?.receiveAddress, buyTxid],
-  )
+  const parsedBuyTxId = useMemo(() => {
+    const isThorTrade = [trade?.sellAsset.assetId, trade?.buyAsset.assetId].includes(
+      thorchainAssetId,
+    )
+    const buyAccountId = isThorTrade ? sellAssetAccountId : buyAssetAccountId
+    const receiveAddress = isThorTrade
+      ? fromAccountId(sellAssetAccountId!).account
+      : trade?.receiveAddress
+    const txId = isThorTrade ? buyTxid.toUpperCase() : buyTxid // Midgard monkey patch Txid is lowercase, but we store Cosmos SDK Txs uppercase
```

Comment:
Since this is already monkey-patched in lib, we might want to respond with the Txid uppercased directly from swapper or this may be a bug waiting to happen